File: base/src/main/java/com/ssz/baselibrary/app/di/component/AppComponent.java
package com.ssz.baselibrary.app.di.component;

import com.ssz.baselibrary.app.AppContext;
import com.ssz.baselibrary.app.di.module.AppModule;
import com.ssz.baselibrary.app.di.module.NetModule;
import com.ssz.baselibrary.model.remote.net.Api;
import com.ssz.baselibrary.model.remote.net.Net;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author : zsp
 * time : 2019 11 2019/11/19 10:58
 */
@Singleton
@Component(modules = {AppModule.class, NetModule.class})
public interface AppComponent {

    AppContext getAppContext();

    Net getNet();

    Api getApi();

    void inject(AppContext appContext);
}

File: app/src/main/java/com/ssz/studydemo/base/dagger/di/component/AppComponent.kt
package com.ssz.studydemo.base.dagger.di.component

import com.ssz.studydemo.base.dagger.di.module.AppModule
import com.ssz.studydemo.base.dagger.di.module.NetModule
import com.ssz.studydemo.model.remote.net.Api
import com.ssz.studydemo.test.app.AppDelegate
import dagger.Component
import javax.inject.Singleton

/**
 * @author : zsp
 * time : 2019 12 2019/12/11 14:32
 */
@Singleton
@Component(modules = [AppModule::class, NetModule::class])
interface AppComponent {
    fun getAppDelegate() : AppDelegate
    fun getApi():Api
    fun inject(appDelegate: AppDelegate)
}